### ISSUE TITLE
[Feature Request] Azure - virtual network gateway and connection resources

### ISSUE THREAD

**Issue by gitmayor** [hashicorp/terraform#10634]
Hi Guys,
Any plans for expanding Azure resources bucket? 

**Terraform Version**
Terraform v0.7.13

**Affected Resource(s)**
- virtual network gateway
- connection

Actually if you want to set site-2-site vpn between any data center IPSec vpn and Azure, you have to create bunch of resources required for that purpose like: 
- local network gateway,
- virtual network gateway,
- public IP for the virtual network gateway,
- connection

All these items can be created either manually or via Microsoft ARM templates. It complicates infrastructure creation and management from Terraform tool. If above resources exist it would be possible to automate vpn creation.

**Comment by TStraub-rms** [comment]
This is something we would love to see as well and would get 100% of our current networking infra into terraform.

**Comment by pmcatominey** [comment]
@gitmayor I've got a work in progress PR #9255

Still waiting on Azure to look into an issue when deleting the Gateway Subnet.

**Comment by pmcatominey** [comment]
Realised that this is a duplicate of #8372.
